src/pubkey.rs

```diff
@@ -0,0 +1,118 @@
+use ark_ff::{BigInteger, PrimeField};
+use bs58;
+use sha2::{Digest, Sha256};
+use std::ops::Neg;
+
+use crate::{FieldHelpers, PallasField, PallasPoint};
+
+const MINA_ADDRESS_LEN: usize = 55;
+
+pub type PubKey = PallasPoint;
+
+#[derive(Clone, Copy)]
+pub struct CompressedPubKey {
+    pub x: PallasField,
+    pub is_odd: bool,
+}
+
+fn to_address(x: PallasField, is_odd: bool) -> String {
+    let mut raw: Vec<u8> = vec![
+        0xcb, // version for base58 check
+        0x01, // non_zero_curve_point version
+        0x01, // compressed_poly version
+    ];
+
+    // pub key x-coordinate
+    raw.extend(x.to_bytes());
+
+    // pub key y-coordinate parity
+    raw.push(is_odd as u8); // TODO: confirm is_even() like this is correct
+
+    // 4-byte checksum
+    let hash = Sha256::digest(&Sha256::digest(&raw[..])[..]);
+    raw.extend(&hash[..4]);
+
+    return bs58::encode(raw).into_string();
+}
+
+impl CompressedPubKey {
+    pub fn to_address(self) -> String {
+        to_address(self.x, self.is_odd)
+    }
+}
+
+pub trait PubKeyHelpers {
```

Comment:
does that need to be a trait?

Reply:
I don't have my own struct for `PubKey`... it's just a `PallasPoint`, which is just a `mina_curves::pasta::pallas:Affine`

Reply:
ah, then I would recommend creating your own struct for a Pubkey, since you usually want to use types to enforce things down the line